### PR TITLE
add support for packaged libvpx (enabled via TG_OWT_VPX_PACKAGED_BUILD)

### PR DESCRIPTION
**Pull request by vitlav**
Just for review. As for me, it works.
In replacement to https://github.com/desktop-app/tg_owt/pull/13

### CODE REVIEW

File: cmake/external.cmake [hunk 1]
```diff
@@ -141,3 +141,34 @@ function(link_x11 target_name)
         target_link_libraries(${target_name} PRIVATE ${X11_Xtst_LIB})
     endif()
 endfunction()
+
+# libvpx
+function(link_vpx target_name)
+    if (TG_OWT_VPX_PACKAGED_BUILD)
```
Comment (ilya-fedin) [inline cmake/external.cmake:147]: I don't think we really need such variable for each library. This was a mess in tdesktop and was removed. Can you use TG_OWT_PACKAGED_BUILD instead?

Comment (ilya-fedin) [reply cmake/external.cmake:147]: Also, I think it's worth to have behavior just like in tdesktop: fallback to the bundled library if packaged one is not found

Comment (ilya-fedin) [reply cmake/external.cmake:147]: An example of how this is done in tdesktop: https://github.com/desktop-app/cmake_helpers/blob/master/external/dbusmenu_qt/CMakeLists.txt#L10-L22

File: cmake/external.cmake [hunk 2]
```diff
@@ -141,3 +141,34 @@ function(link_x11 target_name)
...
+        find_package(PkgConfig REQUIRED)
+        pkg_check_modules(VPX vpx)
+        target_include_directories(${target_name} PRIVATE ${VPX_INCLUDE_DIRS})
+        target_link_libraries(${target_name} PRIVATE ${VPX_LINK_LIBRARIES})
+    else()
```
Comment (ilya-fedin) [inline cmake/external.cmake:152]: Probably, it should be effectively like this
```suggestion
    endif()
    if (NOT VPX_FOUND)
```

**Comment by ilya-fedin** [comment]
Tried to build your PR in various configurations and it sadly still breaks building of snap package just like the previous one
https://github.com/ilya-fedin/tdesktop/runs/1944599327?check_suite_focus=true

**Comment by john-preston** [comment]
#55 introduced packaged build for different third_party dependencies. It is in the `dev` branch right now.

**Pull request closed without merging**